novawallet/Modules/ManualBackupKeyList/ManualBackupKeyListPresenter.swift

```diff
@@ -0,0 +1,200 @@
+        let defaultChainsHeaderText = R.string.localizable.chainAccountsListDefaultHeader(
+            preferredLanguages: localizationManager.selectedLocale.rLanguages
+        )
+
+        let defaultChainsTitleText = R.string.localizable.chainAccountsListDefaultTitle(
+            preferredLanguages: localizationManager.selectedLocale.rLanguages
+        )
+
+        return .defaultKeys(
+            .init(
+                headerText: defaultChainsHeaderText.uppercased(),
+                accounts: [
+                    .init(
+                        title: defaultChainsTitleText,
+                        subtitle: formattedString(for: chains)
+                    )
+                ]
+            )
+        )
+    }
+
+    func createCustomChainsSection(for chains: [ChainModel]) -> ManualBackupKeyListViewLayout.Sections {
+        let customChainsViewModels = chains
+            .compactMap { [weak self] chain -> ManualBackupKeyListViewLayout.CustomAccount? in
+                guard let self else { return .none }
+
+                return ManualBackupKeyListViewLayout.CustomAccount(
+                    network: networkViewModelFactory.createViewModel(from: chain),
+                    chainId: chain.chainId
+                )
+            }
+
+        let customChainsHeaderText = R.string.localizable.chainAccountsListCustomHeader(
+            preferredLanguages: localizationManager.selectedLocale.rLanguages
+        )
+
+        return .customKeys(
+            .init(
+                headerText: customChainsHeaderText.uppercased(),
+                accounts: customChainsViewModels
+            )
+        )
+    }
+
+    func sorted(
+        _ chains: [ChainModel.Id: ChainModel],
+        for metaAccount: MetaAccountModel
+    ) -> SortedChains {
+        let defaultChainsIds = Set(metaAccount.chainAccounts.map(\.chainId))
+
+        var defaultChains: [ChainModel] = []
+        var customChains: [ChainModel] = []
+
+        chains.forEach { chain in
+            defaultChainsIds.contains(chain.key)
+                ? customChains.append(chain.value)
+                : defaultChains.append(chain.value)
+        }
+
+        defaultChains.sort { ChainModelCompator.defaultComparator(chain1: $0, chain2: $1) }
+        customChains.sort { ChainModelCompator.defaultComparator(chain1: $0, chain2: $1) }
+
+        return .init(
+            defaultChains: defaultChains,
+            customChains: customChains
+        )
+    }
+
+    func formattedString(for defaultChains: [ChainModel]) -> String {
+        let chainsToMention = defaultChains.count > 1
+            ? defaultChains.prefix(2)
+            : defaultChains.prefix(1)
+        let separator = ", "
+        let restCount = defaultChains.count - chainsToMention.count
+
+        return chainsToMention
+            .map(\.name)
+            .reduce(into: "") { partialResult, name in
```

Comment:
I think it is too complicated. I think we can first joins chainsToMention name by separator and then check whether we need to append + suffix.